pkg/kubectl/cmd/exec.go

```diff
@@ -161,6 +162,10 @@ func (p *ExecOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, argsIn []
 		}
 	}
 
+	if strings.HasPrefix(p.PodName, "pod/") {
```

Comment:
@deads2k I decided to go the way that kargakis suggested just because the whole infrastructure already exists